Add while(!Serial) for reliable Serial init

polastre opened the pull request:
If you don't include `while(!Serial);`, Clyde's serial
port is very very unreliable.  This is due to the fact
that Clyde is based on the Ardunino Leonardo, which shares
USB with serial communication.

Add a Serial output so people know that Clyde is now ready
and willing to accept commands.

polastre committed:
Add while(!Serial) for reliable Serial init

If you don't include `while(!Serial);`, Clyde's serial
port is very very unreliable.  This is due to the fact
that Clyde is based on the Ardunino Leonardo, which shares
USB with serial communication.

Add a Serial output so people know that Clyde is now ready
and willing to accept commands.

polastre commented:
Now I know why you don't have `while (!Serial) ;` --
- If you have that line in, Clyde won't start executing until the Serial Monitor is opened (bad)
- If you don't have that line, Clyde starts up and runs just fine.

In contrast though:
- If you want to send serial commands to Clyde, you **must** have that line in your file
- If you don't have that line, serial communications are unreliable

I recommend including the line, but commenting it out with a comment that says "Uncomment me for serial communications".  I can make that change to the PR and push it if you approve.

bnadeau commented:
I agree we need the line in there with a comment note about serial communication.

I was wondering if there could be a clever way to check for Serial for a short period of time, or maybe wrap the Serial.print in a function that checks for if(Serial) before printing anything. I haven't thought that through yet.

polastre committed:
Comment out serial ready line for normal operation

Normal operation means not connected to a serial port.

- If you have that line in, Clyde won't start executing until the Serial Monitor is opened (bad)
- If you don't have that line, Clyde starts up and runs just fine.

So, if you want to talk to Clyde,
- If you want to send serial commands to Clyde, you must have that line in your file
- If you don't have that line, serial communications are unreliable

polastre commented:
I've updated the PR to comment out the serial line but include a note to uncomment it to have serial communications with Clyde (see 11a3d9a).
